### PR TITLE
1488 Endo Passable

### PR DESCRIPTION
**Pull request by turadg**
#endo-branch: master 

refs: #1488

## Description

Adapt to actual Passable type

Reviewers, the bar as yet is not whether to merge to this repo. This PR is to serve as evidence that  https://github.com/endojs/endo/pull/2238 can merge and release and agoric-sdk can quickly adopt the new Endo packages.

Once the required Endo code is available by NPM, 
1. bump the Endo packages here to their new releases
2. remove `#endo-branch` pragma
3. verify the Cloudflare job starts passing (it fails now because it doesn't know about #endo-branch)

### Testing Considerations

To test locally,
```
cd $ENDO_REPO
git checkout 1488-Passable-redux
git pull
cd $AGORIC_REPO
scripts/replace-package.sh $ENDO_REPO
```

### CODE REVIEW

File: packages/ERTP/src/types.js
Comment (dckc) [inline packages/ERTP/src/types.js]: that doesn't seem right.

File: packages/zoe/src/zoeService/startInstance.js
Comment (dckc) [inline packages/zoe/src/zoeService/startInstance.js]: in my work-in-progress docs branch, I used `@type ZoeService['startInstance']` here, I think

File: packages/ERTP/src/amountMath.js
Comment (dckc) [inline packages/ERTP/src/amountMath.js]: I wonder if this should take an `unknown`.

File: packages/ERTP/src/amountMath.js
Comment (dckc) [inline packages/ERTP/src/amountMath.js:209]: doesn't `AmountMath.make(aNatBrand, ['a set value'])` falsify this inference?

File: packages/ERTP/src/amountMath.js
Comment (dckc) [inline packages/ERTP/src/amountMath.js]: What are we casting to what?

For each `@ts-expect-error`, I'm trying to understand why we're certain that the declared type is correct.

File: packages/ERTP/src/amountMath.js
Comment (dckc) [inline packages/ERTP/src/amountMath.js:283]: I don't understand this comment.

Perhaps it's more clear if I check out this branch and try removing the `@ts-expect-error`...
Well, no, vs-code says it's not needed.
`yarn lint:types` finds 57 errors in 20 files. I wonder what's going on.

**Review by dckc — COMMENTED** [review]
hope to discuss my comments so far

**Comment by turadg** [comment]
https://github.com/endojs/endo/pull/2238/ merged to master so I updated the `#endo-branch`. I've also rebased onto master and included type coverage before/after measurements.

Once this is green the next steps are to:
- publish latest Endo
- update this PR to include those Endo versions
- get it back to green
- merge

**Comment by dckc** [comment]
odd... commit aefc7cd9f78b44eef7963d9577ff3b8616756543 is empty?

**Comment by turadg** [comment]
@dckc I changed the base to https://github.com/Agoric/agoric-sdk/tree/integration-endo-master so this can merge. Then @kriskowal 's  PR bumping Endo will include those commits

File: packages/zoe/tools/types.js
Comment (dckc) [inline packages/zoe/tools/types.js]: I wonder why this is necessary. it looks like the `TimerService` type already inclues `RemotableObject`.

File: packages/ERTP/src/amountMath.js
Comment (dckc) [inline packages/ERTP/src/amountMath.js]: interesting! Amount values have to be `Key`s. I suppose that constraint was implicit/unstated in a lot of places for a long time.

File: packages/ERTP/src/types.js
Comment (dckc) [inline packages/ERTP/src/types.js]: should be "mint a new payment"
orthogonal to this PR, though

File: packages/smart-wallet/src/offers.js
Comment (dckc) [inline packages/smart-wallet/src/offers.js]: I'm curious why this is necessary.

I'm confident that it's not a problem, though. The whole OfferSpec gets pattern-matched sufficiently to ensure that `offerArgs` is some sort of passable.

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (dckc) [inline packages/inter-protocol/src/provisionPoolKit.js:201]: runtime change... hm...
ok, I guess

File: packages/vats/src/lib-board.js
Comment (dckc) [inline packages/vats/src/lib-board.js:111]: This doesn't seem right. `PassableCap` includes `Promise`, no? Surely keys to a store need to be `Key`s, right?

tests show that board values must be scalar keys - "any Passable primitive value or Remotable".

File: packages/vats/src/vat-bank.js
Comment (dckc) [inline packages/vats/src/vat-bank.js:358]: that seems like a non-trivial API change. Was using promises here already broken?

File: packages/inter-protocol/src/proposals/core-proposal.js
Comment (dckc) [inline packages/inter-protocol/src/proposals/core-proposal.js:30]: yikes!

File: packages/SwingSet/src/devices/vat-admin/device-vat-admin.js
Comment (dckc) [inline packages/SwingSet/src/devices/vat-admin/device-vat-admin.js]: Does this seem correct? If not, should it have XXX?

array destructuring 2 lines down strongly suggests `Array<something>`.
`unknown[]` would be best - i.e. "tsc, please gripe if I have made any assumptions about the contents of this array."

File: packages/SwingSet/src/types-external.js
Comment (dckc) [inline packages/SwingSet/src/types-external.js:358]: ouch. that seems hard on devex

File: packages/base-zone/src/types.js
Comment (dckc) [inline packages/base-zone/src/types.js]: is this an edit of an earlier edit in this PR?
That suggests reviewing by commit (which I'm doing) is not a good idea.

It also suggests squashing. is that the plan?

**Review by dckc — APPROVED** [review]
lots of good stuff!

I'm pretty sure the board change to `PassableCap` is a regression, since it includes `Promise`, which can't be used as a key. I almost requested changes over that. But I'll leave it to your discretion / future work.

File: packages/zoe/tools/types.js
Comment (turadg) [reply packages/zoe/tools/types.js]: I think I needed this in order to to add to the methods.

File: packages/smart-wallet/src/offers.js
Comment (turadg) [reply packages/smart-wallet/src/offers.js]: because of the nesting of `unknown`, making it not qualify as `Passable`.

File: packages/base-zone/src/types.js
Comment (turadg) [reply packages/base-zone/src/types.js]: Plan is not to squash. I could squash some if you advise.

File: packages/smart-wallet/src/offers.js
Comment (dckc) [reply packages/smart-wallet/src/offers.js]: oh! I see.

(thinking out loud): I wonder if `offerArgs?: Passable` would work...
maybe I'll try it after this lands

File: packages/base-zone/src/types.js
Comment (dckc) [reply packages/base-zone/src/types.js]: one idea: smush all the commits together and then split them up by package, in topological order. In each package-commit, describe the changes made to that package.

not that important, but it would be kinda ideal

File: packages/vats/src/vat-bank.js
Comment (turadg) [reply packages/vats/src/vat-bank.js:358]: Yep

File: packages/SwingSet/src/devices/vat-admin/device-vat-admin.js
Comment (turadg) [reply packages/SwingSet/src/devices/vat-admin/device-vat-admin.js]: I time box types improvements in swingset packages because they're usually a rabbit hole

File: packages/base-zone/src/types.js
Comment (turadg) [reply packages/base-zone/src/types.js]: That's a good idea. I've already done a lot of commit grooming on this PR but I'll try to remember to try that next time

File: packages/SwingSet/src/types-external.js
Comment (dckc) [inline packages/SwingSet/src/types-external.js]: stray `)`s:

```suggestion
 * @import {Guarded} from '@endo/exo';
 * @import {Passable, RemotableObject} from '@endo/pass-style';
```

File: packages/SwingSet/src/types-external.js
Comment (turadg) [reply packages/SwingSet/src/types-external.js:358]: that's because the JSDoc is currently on the interface and not the implementation. `VatAdminFacet` is a concrete type so it would be better to document its implementation and then the result would just be typeof the actual facet.

File: packages/SwingSet/src/devices/vat-admin/device-vat-admin.js
Comment (turadg) [reply packages/SwingSet/src/devices/vat-admin/device-vat-admin.js]: This was a vestige of some earlier version of `unserialize`. since `argsCapData` is any `args` is too. removing

File: packages/smart-wallet/src/offers.js
Comment (turadg) [reply packages/smart-wallet/src/offers.js]: it does! I'm force pushing with that update

File: packages/SwingSet/src/types-external.js
Comment (turadg) [reply packages/SwingSet/src/types-external.js]: thanks. I grepped for more and cleaned them up

**Comment by turadg** [comment]
Landed in the sync PR

**Pull request closed without merging**
